base_layer/wallet/src/transaction_service/service.rs

```
@@ -673,6 +694,143 @@ where
+            .map_err(|e| TransactionServiceProtocolError::new(tx_id, TransactionServiceError::from(e)))?;
+        let spending_key = PrivateKey::from_bytes((script_offset_private_key.clone() * dest_pubkey.clone()).as_bytes())
+            .map_err(|e| TransactionServiceProtocolError::new(tx_id, TransactionServiceError::from(e)))?;
+
+        let sender_message = TransactionSenderMessage::new_single_round_message(stp.get_single_round_message()?);
+        let rtp = ReceiverTransactionProtocol::new(
+            sender_message,
+            PrivateKey::random(&mut OsRng),
+            spending_key,
+            OutputFeatures::default(),
+            &self.resources.factories,
+        );
+
+        let recipient_reply = rtp.get_signed_data()?.clone();
+
+        // Start finalizing
+
+        stp.add_single_recipient_info(recipient_reply, &self.resources.factories.range_proof)
+            .map_err(|e| TransactionServiceProtocolError::new(tx_id, TransactionServiceError::from(e)))?;
+        if !stp.is_finalizing() {
+            return Err(TransactionServiceError::OneSidedTransactionError(
+                "Sender protocol single round tx not being finalized".to_string(),
+            ));
+        }
+
+        // Finalize
+
+        stp.finalize(KernelFeatures::empty(), &self.resources.factories)
+            .map_err(|e| {
+                error!(
+                    target: LOG_TARGET,
+                    "Transaction (TxId: {}) could not be finalized. Failure error: {:?}", tx_id, e,
+                );
+                TransactionServiceProtocolError::new(tx_id, TransactionServiceError::from(e))
+            })?;
+        if !stp.is_finalized() {
+            return Err(TransactionServiceError::OneSidedTransactionError(
+                "Sender protocol single round tx not finalized".to_string(),
+            ));
+        }
+        trace!(target: LOG_TARGET, "Finalized one-side transaction TxId: {}", tx_id);
+
+        let _ = self
```

What's the reason for swallowing this error?

---

This event being sent is important, but not critical to the protocol being successful. It could be that the receiver dropped a handle for some reason.

_From the Tokio documentation:_
A return value of Ok does not mean that the sent value will be observed by all or any of the active Receiver handles. Receiver handles may be dropped before receiving the sent message. A return value of Err does not mean that future calls to send will fail.

---

Anything like this that is out of the ordinary should have a comment stating why it can be swallowed, so that new devs don't copy this code thinking it's the norm

---

Agreed, added a proper comment